File: Practica 1/S3/CryptoFilters/FilterInvestment.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   FilterInvestment.cpp
 * Author: reko98
 * 
 * Created on 25 de marzo de 2021, 12:56
 */

#include "FilterInvestment.h"

FilterInvestment::FilterInvestment() {
}

FilterInvestment::FilterInvestment(const FilterInvestment& orig) {
}

FilterInvestment::~FilterInvestment() {
}

FilterInvestment::FilterInvestment(double newValue){
    setCantidadInversion(newValue);
    
}

double FilterInvestment::apply(CryptoCurrency& moneda) {
	double valorPorcentaje;
	double cantidadIncremento;
	cout << "El valor de la inversion que se va a realizar es: " << this->getCantidadInversion() << endl;
	valorPorcentaje = (this->cantidadInversion*100)/moneda.getCapital();

	valorPorcentaje = valorPorcentaje/100;
        
        cout << "El porcentaje que se va a aplicar con el precio es: "<< valorPorcentaje << endl;

	cantidadIncremento=valorPorcentaje*moneda.getValorActual();

	return moneda.getValorActual()+cantidadIncremento;
	
}

File: Practica-2/App/crypto_visionary/lib/FIlterInvestment.dart
import 'package:crypto_visionary/Cryptocurrency.dart';
import 'package:flutter/material.dart';
import 'Filter.dart';

class FilterInvestment implements Filter {

  double cantidadInversion;

  void setCatindadInversion(double nuevaCantidad){
    this.cantidadInversion = nuevaCantidad;
  }

  double getCantidadInversion(){
    return this.cantidadInversion;
  }

  @override
  double apply(Cryptocurrency moneda){
    double valorPorcentaje;
    double cantidadIncremento;

    valorPorcentaje = (this.getCantidadInversion()*100)/moneda.getCapital();
    valorPorcentaje = valorPorcentaje/100;

    cantidadIncremento=valorPorcentaje*moneda.getValorActual();
    return moneda.getValorActual()+cantidadIncremento;
  }




}